Fix normalization of SDTs in table cells

sixlettervariables opened the pull request:
This resolves #172 by pulling the `W.tc`s from their actual parent: `W.sdtContent`.

msftclas commented:
This seems like a small (but important) contribution, so no Contribution License Agreement is required at this point. We will now review your pull request.
_Thanks,
Microsoft Pull Request Bot_

ThomasBarnekow commented:
@sixlettervariables and @EricWhiteDev This looks good to me. My only question would be whether this deserves a unit test.

Obviously, the current and wrong behavior went undetected by the existing unit tests. Thus, a new unit test should establish the correct behavior. This would also make sure it won't be broken in the future.

sixlettervariables commented:
It probably should, however, I've run across a new problem with it (one I fixed by reorganizing the content controls). Sometimes when you nest tables within tables, and have a repeater around the row you get a structure like:

- tr
  - sdt (repeatingSection)
    - sdtContent
      - sdt (repeatingSectionItem)
        - sdtContent
          - tc

This fails, however, the solution isn't really obvious beyond perhaps only running the code if the sdtContent contains a tc. I played with approaches that included rotating nodes in the tree, but none were clean to implement. In any case, I fixed this by reformatting my document.

ThomasBarnekow commented:
@sixlettervariables If the fix as it currently stands solves one issue while possibly not solving others, it is an improvement of the current code. Thus, while we might want to be able to deal with the situation as well, we should likely still go forward with the pull request as soon as you find the time to add a unit test.
